client/src/app/employee/employee-payroll-create/employee-payroll-create.component.ts

```diff
@@ -1,33 +1,35 @@
-import { Component, Input, OnChanges, SimpleChanges } from '@angular/core';
+import { Component, Input } from '@angular/core';
 import { FormControl, FormGroup } from '@angular/forms';
 
 import { EmployeeDetail, EmployeePayrollCreate } from '@employee/employee-detail/state/employee-detail.model';
 import { EmployeeDetailService } from '@employee/employee-detail/state/employee-detail.service';
+import { ClockService } from '@shared/clock.service';
 
 @Component({
   selector: 'app-employee-payroll-create',
   templateUrl: './employee-payroll-create.component.html',
   styleUrls: ['./employee-payroll-create.component.scss'],
 })
-export class EmployeePayrollCreateComponent implements OnChanges {
+export class EmployeePayrollCreateComponent {
   filterForm = new FormGroup({
-    checkDate: new FormControl(''),
+    checkDate: new FormControl(this.clock.now().getDate()),
     employeeId: new FormControl(''),
-    grossPayroll: new FormControl(''),
+    grossPayroll: new FormControl(0),
   });
 
   @Input()
   set employee(employee: EmployeeDetail) {
     this.filterForm.patchValue({
       employeeId: employee.id,
+      checkDate: this.clock.now(),
+      grossPayroll: 0,
     });
   }
 
-  constructor(private detailService: EmployeeDetailService) {}
-
-  ngOnChanges(changes: SimpleChanges): void {
```

Comment:
This was causing the form values to be cleared out because the `@Input()` binding is updated first, then `OnChanges()` runs.

So the `employeeId` value was assigned to the form but then the form was reset and it lost the value, which caused an API request validation error.